BOLT #7: receiving node requirements related to timestamp for channel_update message (#621)
lightningnetwork/lightning-rfc: 07-routing-gossip.md

  - if the specified `chain_hash` value is unknown (meaning it isn't active on
  the specified chain):
    - MUST ignore the channel update.

>   - if `timestamp` is lower than that of the last-received

Author: @cfromknecht Thanks! I included your suggestion for the condition where the `timestamp` is equal. But in order to not leave other holes, I slightly changed it in terms of order than what you suggested. Let me know if its good.

>   - if `timestamp` is NOT greater than OR equal to that of the last-received

Reviewer: Maybe replacing `NOT greater than OR equal to` by `lower than` makes it clearer?

Author: Thanks for the review! I have applied both your changes to your commitment. I'll also change the timestamp config to 'lower than' rather than 'greater than equal to'

  - if the specified `chain_hash` value is unknown (meaning it isn't active on
  the specified chain):
    - MUST ignore the channel update.
  - if the `timestamp` is equal to the last-received `channel_update` for this
    `short_channel_id` AND `node_id`:
  	- if the fields below `timestamp` differ:
		- MAY blacklist this `node_id`.
		- MAY forget all channels associated with it.
	- if the fields below `timestamp` are equal:
		- SHOULD ignore this message	
  - if `timestamp` is lower than that of the last-received
  `channel_update` for this `short_channel_id` AND for `node_id`:
    - SHOULD ignore the message.

>   - otherwise:

Reviewer 2: I guess this otherwise could have been removed too?

Author: @jtimon No that otherwise is needed. Just that the `if` and `otherwise` should be indented on the previous otherwise. I think there was a formatting error when creating the pull request.

Author: Is reverting the commit and making changes the right way to go?

Reviewer: No we don't rewrite history. You can submit a tiny indentation fix PR and it should be approved very quickly.

Author: @t-bast cool. Will submit one soon.

Reviewer: Thanks!

1970-01-01). This cannot be a hard requirement, however, given the possible case
of two `channel_update`s within a single second.

It is assumed that more than one `channel_update` message changing the channel 

> parameters in the same second is a DoS attempt, and therefore, the node responsible 

Reviewer: I'd be less strict on categorizing something as a DoS attempt, you'd need more historical data of a peer misbehaving before putting it in a blacklist.

```suggestion
parameters in the same second may be a DoS attempt, and therefore, the node responsible 
```

> for signing such messages ought to be blacklisted. However, a node can send a same 

Reviewer: ```suggestion
for signing such messages may be blacklisted. However, a node can send a same 
```

> `channel_update` message with a different signature (changing the nonce in signature 

Reviewer 3: important to note that since ECDSA signatures are malleable, the signatures can be different even if the node only signed once.

Author: Very fair point. I'm incorporating this in the rationale.
